### PR TITLE
assertCompletesM

### PR DESCRIPTION
**Pull request by emarx**
This PR adds a function `assertCompletesM` to complement the existing `assertCompletes`

### CODE REVIEW

**Comment by regiskuckaertz** [comment]
@emarx you will need to run `sbt prepare` to fix all those lint issues.

File: test/shared/src/main/scala/zio/test/package.scala [hunk 1]
```diff
@@ -171,6 +171,12 @@ package object test extends CompileVariants {
   val assertCompletes: TestResult =
     assertImpl(true)(Assertion.isTrue)
 
+  /**
+   * Asserts that the given test was completed.
+   */
+  val assertCompletesM: UIO[TestResult] =
+    assertMImpl(UIO(true))(Assertion.isTrue)
```
Comment (regiskuckaertz) [inline test/shared/src/main/scala/zio/test/package.scala:178]:
```suggestion
    assertMImpl(UIO.succeed(true))(Assertion.isTrue)
```
sorry to be pedantic, `UIO(x)` is when `x` is an effect.

**Comment by emarx** [comment]
Thanks, @regiskuckaertz! Ran `sbt publish` and made the requested change.

File: test/shared/src/main/scala/zio/test/package.scala [hunk 2]
```diff
@@ -171,6 +171,12 @@ package object test extends CompileVariants {
   val assertCompletes: TestResult =
     assertImpl(true)(Assertion.isTrue)
 
+  /**
+   * Asserts that the given test was completed.
+   */
+  val assertCompletesM: UIO[TestResult] =
+    assertMImpl(UIO.succeed(true))(Assertion.isTrue)
```
Comment (adamgfraser) [inline test/shared/src/main/scala/zio/test/package.scala:178]: Can actually be `succeedNow` for internal implementations like this.

File: test/shared/src/main/scala/zio/test/package.scala [hunk 3]
```diff
@@ -656,6 +662,7 @@ package object test extends CompileVariants {
   }
 
   object Sized {
+
```
Comment (adamgfraser) [inline test/shared/src/main/scala/zio/test/package.scala:665]: Unrelated changes?

**Review by adamgfraser — CHANGES_REQUESTED** [review]
Looks great! Just a couple of very minor comments.

File: test/shared/src/main/scala/zio/test/package.scala (hunk 3 above)
Comment (emarx) [reply test/shared/src/main/scala/zio/test/package.scala]: Whoops, hadn't switched the scalafmt

**Comment by emarx** [comment]
Thanks for your review, @adamgfraser! Comments addressed.

**Review by adamgfraser — APPROVED** [review]

**Comment by adamgfraser** [comment]
Looks great! Will merge after all the checks pass.

**Pull request merged**